app/javascript/packs/base.jsx

```diff
@@ -144,3 +144,25 @@ getInstantClick().then((spa) => {
 });
 
 initializeNav();
+
+async function loadCreatorSettings() {
+  try {
+    const [{ CreatorSettingsController }, { Application }] = await Promise.all([
+      import('@controllers/creator_settings_controller'),
+      import('@hotwired/stimulus'),
+    ]);
+
+    const application = Application.start();
+    application.register('creator-settings', CreatorSettingsController);
+  } catch (error) {
+    Honeybadger.notify(
+      `Error loading the creator settings controller: ${error.message}`,
+    );
+  } finally {
+    // TODO: Friendly error message if the controller fails to load
+  }
+}
+
+if (document.location.pathname === '/admin/creator_settings/new') {
```

Comment:
@Ridhwana, this works better so  that there's no need to worry about whether or not a querystring is present.

Reply:
Looks good @nickytonline , also shall we user routes.js?

```
import { newAdminCreatorSettingPath } from 'routes.js.erb';

if (document.location.pathname === newAdminCreatorSettingPath()) {
  loadCreatorSettings();
}
```

Reply:
Looks good @nickytonline , also shall we user routes.js?

```
import { newAdminCreatorSettingPath } from 'routes.js.erb';

if (document.location.pathname === newAdminCreatorSettingPath()) {
  loadCreatorSettings();
}
```